wasm_memsafety_rtlib.c

```diff
@@ -1,15 +1,19 @@
 #include <stdlib.h>
 #include <stdbool.h>
 #include <stdio.h>
+#include <string.h>
 
 #define TABLE_MAX_LOAD 0.75
 
 typedef struct {
     void *key;
-    size_t size;
+    // The memory region size a user requests.
+    size_t requested_size;
+    // The memory region size a that is actually allocated internally, which
+    // may be larger than requested_size due to alignment constraints.
+    size_t allocated_size;
```

Comment:
I personally think keeping it this way makes sense from a readability perspective, and it does add the small optimization that we only have to copy `requested_size` bytes with `memcpy` in `realloc`, though this would at most save us 15 bytes in the best case.
However, I do acknowledge that this might have a memory overhead, since we need to save this additional `size_t` for each entry. We could turn it into `allocated_size-requested_size` as well, which could be a `uint8_t` that is at most equal to 15, but I'm not sure if this actually improves anything due to padding and hardware specific int sizes.